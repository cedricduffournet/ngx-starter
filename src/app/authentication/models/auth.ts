export interface AuthToken {
  access_token: string;
  expires_in: number;
  token_type: string;
  refresh_token: string;
}

export interface Credentials {
  login: string;
  password: string;
}

export interface ResetPassword {
  plainPassword: {
    first: string;
    second: string;
  };
}

export interface Register extends ResetPassword {
  civility: number;
  firstname: string;
  lastname: string;
  email: string;
  confirmationUrl: string;
}

export interface SendResetPassword {
  username: string;
}
