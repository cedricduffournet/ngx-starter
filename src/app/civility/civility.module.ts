import { NgModule, ComponentFactoryResolver } from '@angular/core';

import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { SharedModule } from '@app/shared/shared.module';
import { CivilityEffects } from '@app/civility/state/effects';
import {
  CivilityAddComponent,
  CivilityUpdateComponent,
  CivilityDeleteComponent,
  CivilityFormComponent,
  CivilityItemComponent,
  CivilityItemsComponent
} from '@app/civility/components';
import {
  CivilityAddModalComponent,
  CivilityUpdateModalComponent,
  CivilityDeleteModalComponent,
  CivilityListViewComponent
} from '@app/civility/containers';
import { reducers } from '@app/civility/state/reducers';
import { CoalescingComponentFactoryResolver } from '@app/coalescing-component-factory-resolver.service';

@NgModule({
  imports: [
    RouterModule,
    SharedModule,
    EffectsModule.forFeature([CivilityEffects]),
    StoreModule.forFeature('civilities', reducers)
  ],
  declarations: [
    CivilityListViewComponent,
    CivilityAddComponent,
    CivilityUpdateComponent,
    CivilityDeleteComponent,
    CivilityFormComponent,
    CivilityItemComponent,
    CivilityItemsComponent,
    CivilityAddModalComponent,
    CivilityUpdateModalComponent,
    CivilityDeleteModalComponent
  ],
  entryComponents: [
    CivilityUpdateModalComponent,
    CivilityAddModalComponent,
    CivilityDeleteModalComponent
  ],
  exports: [CivilityListViewComponent]
})
export class CivilityModule {
  // see https://github.com/angular/angular/issues/14324
  constructor(
    coalescingResolver: CoalescingComponentFactoryResolver,
    localResolver: ComponentFactoryResolver
  ) {
    coalescingResolver.registerResolver(localResolver);
  }
}
