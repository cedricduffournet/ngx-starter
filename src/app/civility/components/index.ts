export * from './civility-add/civility-add.component';
export * from './civility-update/civility-update.component';
export * from './civility-delete/civility-delete.component';
export * from './civility-form/civility-form.component';
export * from './civility-item/civility-item.component';
export * from './civility-items/civility-items.component';
