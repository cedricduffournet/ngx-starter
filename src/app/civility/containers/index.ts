export * from './civility-add-modal/civility-add-modal.component';
export * from './civility-update-modal/civility-update-modal.component';
export * from './civility-delete-modal/civility-delete-modal.component';
export * from './civility-list-view/civility-list-view.component';
