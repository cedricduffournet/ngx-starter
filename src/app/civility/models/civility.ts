import { schema } from 'normalizr';

export interface Civility {
  id: number;
  name: string;
  code: string;
}

export const civilitySchema = new schema.Entity('civilities');
