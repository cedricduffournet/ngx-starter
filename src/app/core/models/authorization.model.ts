export interface Authorization {
  create: boolean;
  update: boolean;
  delete: boolean;
}
