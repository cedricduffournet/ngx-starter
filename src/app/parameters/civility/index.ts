export * from './parameter-civility.component';
