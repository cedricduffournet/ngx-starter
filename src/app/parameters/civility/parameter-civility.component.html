<app-civility-list-view></app-civility-list-view>
