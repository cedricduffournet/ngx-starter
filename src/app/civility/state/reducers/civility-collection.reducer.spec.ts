import { reducer } from '@app/civility/state/reducers/civility-collection.reducer';
import * as fromCivilities from '@app/civility/state/reducers/civility-collection.reducer';

import {
  CivilityListViewActions,
  CivilityAddModalActions,
  CivilityDeleteModalActions,
  CivilityApiActions
} from '@app/civility/state/actions';
import { Civility } from '@app/civility/models/civility';

describe('CivilityCollectionReducer', () => {
  describe('undefined action', () => {
    it('should return the default state', () => {
      const result = reducer(undefined, {} as any);

      expect(result).toMatchSnapshot();
    });
  });

  describe('LOAD', () => {
    it('should set loading to true', () => {
      const action = CivilityListViewActions.loadCivilities();
      const result = reducer(fromCivilities.INITIAL_STATE, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('LOAD_SUCCESS ', () => {
    it('should set loading to false, loaded to true and set civility ids [1, 2]', () => {
      const initialState = {
        ...fromCivilities.INITIAL_STATE,
        loading: true
      };

      const civilities = {
        entities: {
          1: {
            code: 'Mr',
            name: 'Mister'
          },
          2: {
            code: 'Mr',
            name: 'Miss'
          }
        },
        result: [1, 2]
      };
      const action = CivilityApiActions.loadCivilitySuccess({
        civilities
      });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('LOAD_FAILURE ', () => {
    const initialState = {
      ...fromCivilities.INITIAL_STATE,
      loading: true
    };

    it('should set loading to false, loaded to false', () => {
      const error = {
        message: 'error'
      };
      const action = CivilityApiActions.loadCivilityFailure({ error });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('ADD', () => {
    it('should set adding to true on CivilityAddModalActions.addCivility', () => {
      const civility = {
        code: 'AddCode',
        name: 'AddName'
      } as Civility;
      const action = CivilityAddModalActions.addCivility({
        civility
      });
      const result = reducer(fromCivilities.INITIAL_STATE, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('ADD_SUCCESS', () => {
    it('should set adding to false, added to true and add new civility (3) to ids', () => {
      const initialState = {
        ...fromCivilities.INITIAL_STATE,
        adding: true,
        ids: [1, 2]
      };
      const civility = {
        entities: {
          3: {
            code: 'AddCode',
            name: 'AddName'
          }
        },
        result: 3
      };
      const action = CivilityApiActions.addCivilitySuccess({
        civility
      });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('ADD_FAILURE & SHOW_MODAL_ADD', () => {
    const initialState = {
      ...fromCivilities.INITIAL_STATE,
      adding: true,
      added: true
    };

    it('should set adding to false, added to false', () => {
      const error = {
        message: 'error'
      };
      const action = CivilityApiActions.addCivilityFailure({ error });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });

    it('should set adding to false on CivilityListViewActions.showAddCivilityModal', () => {
      const action = CivilityListViewActions.showAddCivilityModal();
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('DELETE', () => {
    it('should set deleting to true', () => {
      const civility = {
        id: 1,
        code: 'RemoveCode',
        name: 'RemoveName'
      };
      const action = CivilityDeleteModalActions.deleteCivility({
        civility
      });
      const result = reducer(fromCivilities.INITIAL_STATE, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('DELETE_SUCCESS', () => {
    const initialState = {
      ...fromCivilities.INITIAL_STATE,
      removing: true,
      ids: [1, 2]
    };

    it('should set deleting to false, deleted to true and remove civility (1) form ids', () => {
      const id = 1;

      const action = CivilityApiActions.deleteCivilitySuccess({
        id
      });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('DELETE_FAILURE & SHOW_MODAL_DELETE', () => {
    const initialState = {
      ...fromCivilities.INITIAL_STATE,
      removing: true,
      removed: true
    };

    it('should set removing to false, remove to false', () => {
      const error = {
        message: 'error'
      };
      const action = CivilityApiActions.deleteCivilityFailure({ error });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });

    it('should set removing to false, remove to false on CivilityListViewActions.showDeleteCivilityModal', () => {
      const action = CivilityListViewActions.showDeleteCivilityModal();
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('Selectors', () => {
    describe('getIds', () => {
      it('should retrieve ids [1,2,3]', () => {
        const result = fromCivilities.getIds({
          ...fromCivilities.INITIAL_STATE,
          ids: [1, 2, 3]
        });
        expect(result).toMatchSnapshot();
      });
    });

    describe('getDeleting', () => {
      it('should retrieve removing false', () => {
        const result = fromCivilities.getDeleting({
          ...fromCivilities.INITIAL_STATE,
          deleting: false
        });

        expect(result).toMatchSnapshot();
      });

      it('should retrieve removing true', () => {
        const result = fromCivilities.getDeleting({
          ...fromCivilities.INITIAL_STATE,
          deleting: true
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getDeleted', () => {
      it('should retrieve deleted false', () => {
        const result = fromCivilities.getDeleted({
          ...fromCivilities.INITIAL_STATE,
          deleted: false
        });

        expect(result).toMatchSnapshot();
      });

      it('should retrieve deleted true', () => {
        const result = fromCivilities.getDeleted({
          ...fromCivilities.INITIAL_STATE,
          deleted: true
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getAdding', () => {
      it('should retrieve adding false', () => {
        const result = fromCivilities.getAdding({
          ...fromCivilities.INITIAL_STATE,
          adding: false
        });

        expect(result).toMatchSnapshot();
      });

      it('should retrieve adding true', () => {
        const result = fromCivilities.getAdding({
          ...fromCivilities.INITIAL_STATE,
          adding: true
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getAdded', () => {
      it('should retrieve added false', () => {
        const result = fromCivilities.getAdded({
          ...fromCivilities.INITIAL_STATE,
          added: false
        });

        expect(result).toMatchSnapshot();
      });

      it('should retrieve added true', () => {
        const result = fromCivilities.getAdded({
          ...fromCivilities.INITIAL_STATE,
          added: true
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getLoaded', () => {
      it('should retrieve loaded false', () => {
        const result = fromCivilities.getLoaded({
          ...fromCivilities.INITIAL_STATE,
          loaded: false
        });

        expect(result).toMatchSnapshot();
      });

      it('should retrieve loaded true', () => {
        const result = fromCivilities.getLoaded({
          ...fromCivilities.INITIAL_STATE,
          loaded: true
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getLoading', () => {
      it('should retrieve loading false', () => {
        const result = fromCivilities.getLoading({
          ...fromCivilities.INITIAL_STATE,
          loading: false
        });

        expect(result).toMatchSnapshot();
      });

      it('should retrieve loading true', () => {
        const result = fromCivilities.getLoading({
          ...fromCivilities.INITIAL_STATE,
          loading: true
        });

        expect(result).toMatchSnapshot();
      });
    });
  });
});
