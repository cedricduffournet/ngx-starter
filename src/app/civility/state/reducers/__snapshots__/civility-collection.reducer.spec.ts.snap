// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CivilityCollectionReducer ADD should set adding to true on CivilityAddModalActions.addCivility 1`] = `
Object {
  "added": false,
  "adding": true,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
}
`;

exports[`CivilityCollectionReducer ADD_FAILURE & SHOW_MODAL_ADD should set adding to false on CivilityListViewActions.showAddCivilityModal 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
}
`;

exports[`CivilityCollectionReducer ADD_FAILURE & SHOW_MODAL_ADD should set adding to false, added to false 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
}
`;

exports[`CivilityCollectionReducer ADD_SUCCESS should set adding to false, added to true and add new civility (3) to ids 1`] = `
Object {
  "added": true,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [
    1,
    2,
    3,
  ],
  "loaded": false,
  "loading": false,
}
`;

exports[`CivilityCollectionReducer DELETE should set deleting to true 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": true,
  "ids": Array [],
  "loaded": false,
  "loading": false,
}
`;

exports[`CivilityCollectionReducer DELETE_FAILURE & SHOW_MODAL_DELETE should set removing to false, remove to false 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
  "removed": true,
  "removing": true,
}
`;

exports[`CivilityCollectionReducer DELETE_FAILURE & SHOW_MODAL_DELETE should set removing to false, remove to false on CivilityListViewActions.showDeleteCivilityModal 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
  "removed": true,
  "removing": true,
}
`;

exports[`CivilityCollectionReducer DELETE_SUCCESS should set deleting to false, deleted to true and remove civility (1) form ids 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": true,
  "deleting": false,
  "ids": Array [
    2,
  ],
  "loaded": false,
  "loading": false,
  "removing": true,
}
`;

exports[`CivilityCollectionReducer LOAD should set loading to true 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": true,
}
`;

exports[`CivilityCollectionReducer LOAD_FAILURE  should set loading to false, loaded to false 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
}
`;

exports[`CivilityCollectionReducer LOAD_SUCCESS  should set loading to false, loaded to true and set civility ids [1, 2] 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [
    1,
    2,
  ],
  "loaded": true,
  "loading": false,
}
`;

exports[`CivilityCollectionReducer Selectors getAdded should retrieve added false 1`] = `false`;

exports[`CivilityCollectionReducer Selectors getAdded should retrieve added true 1`] = `true`;

exports[`CivilityCollectionReducer Selectors getAdding should retrieve adding false 1`] = `false`;

exports[`CivilityCollectionReducer Selectors getAdding should retrieve adding true 1`] = `true`;

exports[`CivilityCollectionReducer Selectors getDeleted should retrieve deleted false 1`] = `false`;

exports[`CivilityCollectionReducer Selectors getDeleted should retrieve deleted true 1`] = `true`;

exports[`CivilityCollectionReducer Selectors getDeleting should retrieve removing false 1`] = `false`;

exports[`CivilityCollectionReducer Selectors getDeleting should retrieve removing true 1`] = `true`;

exports[`CivilityCollectionReducer Selectors getIds should retrieve ids [1,2,3] 1`] = `
Array [
  1,
  2,
  3,
]
`;

exports[`CivilityCollectionReducer Selectors getLoaded should retrieve loaded false 1`] = `false`;

exports[`CivilityCollectionReducer Selectors getLoaded should retrieve loaded true 1`] = `true`;

exports[`CivilityCollectionReducer Selectors getLoading should retrieve loading false 1`] = `false`;

exports[`CivilityCollectionReducer Selectors getLoading should retrieve loading true 1`] = `true`;

exports[`CivilityCollectionReducer undefined action should return the default state 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
}
`;
